File: mobile/src/main/java/com/keepnotes/viewmodels/DetailViewModel.java
package com.keepnotes.viewmodels;

import com.keepnotes.models.Note;
import com.keepnotes.storage.IDataManager;

public class DetailViewModel {
    private Note note;
    private IDataManager manager;

    public DetailViewModel(Note note, IDataManager manager) {
        this.note = note;
        this.manager = manager;
    }

    public Note getNote() {
        return note;
    }

    public String getTitle() {
        return note.getTitle();
    }

    public void removeNote() {
        manager.removeNote(note);
    }

    public void saveNote() {
        manager.saveNote(note);
    }

    public void changeStatus(boolean done) {
        manager.changeNoteStatus(note);
    }
}

File: mobile/src/main/java/com/keepnotes/viewmodels/DetailViewModel.kt
package com.keepnotes.viewmodels

import com.keepnotes.models.Note
import com.keepnotes.storage.IDataManager

class DetailViewModel(val note: Note, private val manager: IDataManager) {

    val title: String
        get() = note.title

    fun removeNote() {
        manager.removeNote(note)
    }

    fun saveNote() {
        manager.saveNote(note)
    }

    fun changeStatus(done: Boolean) {
        manager.changeNoteStatus(note)
    }
}
